src/trsp/GraphDefinition.cpp

```diff
@@ -482,7 +482,7 @@ int GraphDefinition:: my_dijkstra3(
 
         *path = reinterpret_cast<path_element_tt *>(
                 malloc(sizeof(path_element_tt) * (m_vecPath.size() + 1)));
-        *path_count = static_cast<int>(m_vecPath.size());
+        *path_count = m_vecPath.size();
 
         for (size_t i = 0; i < *path_count; i++) {
```

Comment:
hmm, `i` shadows the previous declaration on line 385, FYI.

Reply:
Right, `trsp` code is very ugly. and trsp function has zillions of issues:
https://docs.pgrouting.org/3.1/en/pgr_trsp.html#known-issues
I fixed the warnings detected by clang compiler on ubuntu 20.04 (my computer) 18.04 on actions.
line 385 was not detected.